yellowbrick/features/projection.py

```diff
@@ -0,0 +1,303 @@
+    multi dimensional data to either two or three dimensional space. The transformer
+    which converts multi-dimensional data must  be specified in the subclasses.
+
+    Parameters
+    ----------
+    ax : matplotlib Axes, default: None
+        The axes to plot the figure on. If None is passed in the current axes.
+        will be used (or generated if required).
+
+    features: list, default: None
+        a list of feature names to use
+        If a DataFrame is passed to fit and features is None, feature
+        names are selected as the columns of the DataFrame.
+
+    classes: list, default: None
+        a list of class names for the legend
+        If classes is None and a y value is passed to fit then the classes
+        are selected from the target vector.
+
+    color : list or tuple of colors, default: None
+        Specify the colors for each individual class.
+
+    colormap : string or cmap, default: None
+        Optional string or matplotlib cmap to colorize points.
+        Use either color to colorize the points on a per class basis or
+        colormap to color them on a continuous scale.
+
+    target_type : str, default: "auto"
+        Specify the type of target as either "discrete" (classes) or "continuous"
+        (real numbers, usually for regression). If "auto", then it will
+        attempt to determine the type by counting the number of unique values.
+
+    projection : int or string, default: 2
+        Dimension of the Projection Visualizer.
+
+    alpha : float, default: 0.75
+        Specify a transparency where 1 is completely opaque and 0 is completely
+        transparent. This property makes densely clustered points more visible.
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+    """
+    
+    def __init__(self, ax=None, features=None, classes=None, color=None,
+             colormap=palettes.DEFAULT_SEQUENCE, target_type="auto", projection=2,
+             alpha=0.75, **kwargs):
+
+        super(ProjectionVisualizer, self).__init__(ax=ax, features=features, 
+                                                     classes=classes, color=color,
+                                                     colormap=colormap, 
+                                                     target_type=target_type, **kwargs)
+        # Convert string to integer
+        if isinstance(projection, str):
+            if projection in {'2D', '2d'}:
+                projection = 2
+            if projection in {'3D', '3d'}:
+                projection = 3
+        if projection not in {2, 3}:
+            raise YellowbrickValueError("Projection dimensions must be either 2 or 3")
+        self.projection = projection
+
+        if self.ax.name!='3d' and self.projection == 3:
+                warnings.warn("Uses third feature space as size. Pass 3d axes.", 
```

Comment:
I like very much putting the warning in `__init__` and I'm glad you found a way to test if the axes was 3D - I had no idea that axes had a name property! 

```suggestion
                warnings.warn("data projection to 3 dimensions requires a 3d axes to draw on.", 
```